cmd/dbc/auth.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2025 Columnar Technologies Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/cli/oauth/device"
+	"github.com/columnar-tech/dbc/auth"
+)
+
+type AuthCmd struct {
+	Login  *LoginCmd  `arg:"subcommand" help:"Authenticate with a driver index"`
+	Logout *LogoutCmd `arg:"subcommand" help:"Logout from a driver index"`
+}
+
+type LoginCmd struct {
+	URI      string `arg:"positional" help:"URL of the driver index to authenticate with"`
```

Comment:
I would be really happy if we called this URL instead of URI. Or hostname if we will only ever support top level registries. Going more specific has some advantages for users who are punching in their own info.